### PR TITLE
BREAKING: add new classes for vulnerabilities

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR introduces new classes, `vuln-os-pkgs` and `vuln-lang-pkgs`. `Packages` and `Vulnerabilities` are stored separately.

### Before
There is a single target including both packages and vulnerabilities.

```
{
	"Target": "rails:latest (centos 8.3.2011)",
	"Class":  "os-pkgs",
	"Type":   "centos",
	"Packages": {
		{
			"Name":            "binutils",
			"Version":         "2.30",
			"Release":         "93.el8",
			"Epoch":           0,
			"Arch":            "aarch64",
			"Licenses":       ["GPLv3+"],
		}
	},
	"Vulnerabilities": {
		{
			"VulnerabilityID":  "CVE-2018-20623",
			"PkgName":          "binutils",
			"InstalledVersion": "2.30-93.el8",
			...
		}
	}
}
```

### After
There are two targets for packages and vulnerabilities.

```
{
	"Target": "rails:latest (centos 8.3.2011)",
	"Class":  "os-pkgs",
	"Type":   "centos",
	"Packages": {
		{
			"Name":            "binutils",
			"Version":         "2.30",
			"Release":         "93.el8",
			"Epoch":           0,
			"Arch":            "aarch64",
			"Licenses":       ["GPLv3+"],
		}
	}
},
{
	"Target": "rails:latest (centos 8.3.2011)",
	"Class":  "vuln-os-pkgs",
	"Type":   "centos",
	"Vulnerabilities": {
		{
			"VulnerabilityID":  "CVE-2018-20623",
			"PkgName":          "binutils",
			"InstalledVersion": "2.30-93.el8",
			...
		}
	}
}
```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/report/table/table.go [hunk 1]
```diff
@@ -52,26 +53,33 @@ type Renderer interface {
...
+	}
+
 	for _, result := range report.Results {
 		// Not display a table of custom resources
 		if result.Class == types.ClassCustom {
 			continue
 		}
-		tw.write(result)
+		tw.write(result, pkgs)
 	}
```
Comment (DmitriyLewen) [inline pkg/report/table/table.go:69]: 2 identical loops are performed here.
perhaps we can use 1 loop and store `pkgs` inside loop. We can also use slice(`pkgs :=[]ftypes.Package{}`)
What do you think?

File: pkg/flag/options.go [hunk 2]
```diff
@@ -83,6 +85,20 @@ type Options struct {
...
 
+// Align takes consistency of options
+func (o *Options) Align() {
+	if o.Format == report.FormatSPDX || o.Format == report.FormatSPDXJSON {
+		log.Logger.Debug(`"--format spdx" and "--format spdx-json" disable security checks`)
+		o.SecurityChecks = nil
+	}
+
+	// Vulnerability scanning is disabled by default for CycloneDX.
+	if !viper.IsSet(SecurityChecksFlag.ConfigName) && o.Format == report.FormatCycloneDX {
```
Comment (afdesk) [inline pkg/flag/options.go:96]: it doesn't matter, just a note:
i'd like to change these statements:
```suggestion
	if o.Format == report.FormatCycloneDX && !viper.IsSet(SecurityChecksFlag.ConfigName) {
```

**Review by afdesk — APPROVED** [review]

File: pkg/report/table/table.go [hunk 3]
```diff
@@ -52,26 +53,33 @@ type Renderer interface {
...
 // Write writes the result on standard output
 func (tw Writer) Write(report types.Report) error {
+	pkgs := map[string][]ftypes.Package{}
+	for _, result := range report.Results {
+		if result.Class == types.ClassOSPkg || result.Class == types.ClassLangPkg {
+			pkgs[result.Target] = result.Packages
+		}
+	}
+
 	for _, result := range report.Results {
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/report/table/table.go:63]: No, we have to take packages from `report.Results` before writing the output such as table and json.

File: docs/docs/sbom/index.md [hunk 4]
```diff
@@ -9,9 +9,10 @@ Trivy can generate the following SBOM formats.
 To generate SBOM, you can use the `--format` option for each subcommand such as `image` and `fs`.
 
 ```
-$ trivy image --format cyclonedx --output result.json alpine:3.15
+$ trivy image --format spdx --output result.json alpine:3.15
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/sbom/index.md]: Nice catch. Fixed.
https://github.com/aquasecurity/trivy/pull/2541/commits/164da35655496ffb581ff4f7d9b3621e8979be92

File: pkg/flag/options.go (hunk 2 above)
Comment (knqyf263) [reply pkg/flag/options.go]: Fixed
https://github.com/aquasecurity/trivy/pull/2541/commits/7a885bc4dcc6ea49a1929ef7fbc17519140bf8be

File: pkg/report/table/table.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/report/table/table.go:64]: added a comment
https://github.com/aquasecurity/trivy/pull/2541/commits/1ed1d28ce127e493ed489109788bd72acaa90f9c

**Pull request merged**